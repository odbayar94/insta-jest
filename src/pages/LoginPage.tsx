import React, { useEffect,useState } from 'react';
import { Redirect } from 'react-router-dom';
import { Button } from 'reactstrap';
import Cookies from "js-cookie";

import classNames from "classnames";

function LoginPage(){
 
  return (
    <div className="App">
      <header className="App-header">
       
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <Button label="Hello"></Button>
      </header>
   
    </div>
  );
}

export default LoginPage;