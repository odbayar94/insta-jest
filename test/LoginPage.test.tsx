/**
 * @jest-environment jsdom
 */



import React from 'react';
import { render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider, useDispatch } from 'react-redux'
import configureStore from 'redux-mock-store'

// My own import
import LoginPage from '../src/pages/LoginPage';

describe('LoginPage', () => {
  const initialState = {
          currentUser: null,
          logginIn: false,
          error: false,
          errorMessage: null,
          userId: null,
          token:null,
  }
  const mockStore = configureStore()
  let store: any;

  it('Snap shot"', () => {
    store = mockStore(initialState)
    // const { getByText } = render(<Provider store={store}><LoginPage /></Provider>)
    // expect(getByText('Hello Worldd!')).not.toBeNull()

    const appSnap = renderer.create(<Provider store={store}><LoginPage /></Provider>).toJSON()
    expect(appSnap).toMatchSnapshot()

   
  })
})

// const ReduxProvider = ({ children, reduxStore }) => (
//   <Provider store={reduxStore}>{children}</Provider>
// );

// test("...", () => {
//   const store = configureStore();
//   const wrapper = ({ children }) => (
//     <ReduxProvider reduxStore={store}>{children}</ReduxProvider>
//   );
//   const { result } = renderHook(() => {
//     useSaveAuthenticationDataToStorages(useDispatch());
//   }, { wrapper });
//   // ... Rest of the logic
// });

// jest.mock('react-redux', () => {
//   const ActualReactRedux = require.requireActual('react-redux');
//   return {
//       ...ActualReactRedux,
//       useSelector: jest.fn().mockImplementation(() => {
//           return mockState;
//       }),
//   };
// });




// function useSaveAuthenticationDataToStorages(arg0: any) {
//   throw new Error('Function not implemented.');
// }

// function renderHook(arg0: () => void, arg1: { wrapper: ({ children }: { children: any; }) => JSX.Element; }): { result: any; } {
//   throw new Error('Function not implemented.');
// }

// it("LoginPage maches snapshot", ()=>{
//   const appSnap = renderer.create(<LoginPage/>).toJSON()
//   expect(appSnap).toMatchSnapshot()
// });

// describe('With React Testing Library', () => {
//   const initialState = {output:10}
//   const mockStore = configureStore()
//   let store,wrapper;

//   it('Shows "Hello world!"', () => {
//     store = mockStore(initialState)
//     const { getByText } = render(<Provider store={store}><LoginPage /></Provider>)

//     expect(getByText('Hello Worldd!')).not.toBeNull()
//   })
// })

// jest.mock('react-redux', () => {
//   const ActualReactRedux = jest.requireActual('react-redux');
//   return {
//       ...ActualReactRedux,
//       useSelector: jest.fn().mockImplementation(() => {
//           return mockState;
//       }),
//   };
// });