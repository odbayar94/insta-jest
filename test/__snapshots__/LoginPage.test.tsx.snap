// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Snap shot" 1`] = `
<div
  className="login"
>
  <div
    className="login__image-section"
  >
    <div
      className="login__iphone-header"
    >
      <div
        className="login__iphone-image"
      >
        <img
          alt=""
          className="RP4i1 JtrJi "
          src="https://www.instagram.com/static/images/homepage/screenshot1.jpg/d6bf0c928b5a.jpg"
        />
      </div>
    </div>
    <div
      className="login__section"
    >
      <div
        className="login__container"
      >
        <div
          className="login__form"
        >
          <div
            className="login__header"
          >
            <h1
              className="login__header-img login__header-text"
            >
              Instagram
            </h1>
          </div>
          <form>
            <div
              className="login__username"
            >
              <label>
                <input
                  className="login__username-text"
                  id="username"
                  name="name"
                  onChange={[Function]}
                  placeholder="Phone number, username, or email"
                  type="text"
                />
              </label>
            </div>
            <div
              className="login__password"
            >
              <label
                className="login__password-label"
              >
                <input
                  className="login__password-text"
                  name="name"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </label>
            </div>
            <div
              className="login__submit"
            >
              <label>
                <button
                  aria-label={null}
                  className="login__submit-btn login__submit-btn-active btn btn-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Login In
                </button>
              </label>
            </div>
          </form>
          <div
            className="login__forget"
          >
            <div
              className="login__line"
            >
              <div
                className="login__line-abstract"
              />
              <div
                className="login__text-or"
              >
                or
              </div>
              <div
                className="login__line-abstract"
              />
            </div>
            <div
              className="login__forget-text"
            >
              Forgot password?
            </div>
          </div>
        </div>
        <div
          className="login__signup"
        >
          <p>
            Don't have an account? Sign up
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`With React Testing Library Shows "Hello world!" 1`] = `
<div
  className="login"
>
  <div
    className="login__image-section"
  >
    <div
      className="login__iphone-header"
    >
      <div
        className="login__iphone-image"
      >
        <img
          alt=""
          className="RP4i1 JtrJi "
          src="https://www.instagram.com/static/images/homepage/screenshot1.jpg/d6bf0c928b5a.jpg"
        />
      </div>
    </div>
    <div
      className="login__section"
    >
      <div
        className="login__container"
      >
        <div
          className="login__form"
        >
          <div
            className="login__header"
          >
            <h1
              className="login__header-img login__header-text"
            >
              Instagram
            </h1>
          </div>
          <form>
            <div
              className="login__username"
            >
              <label>
                <input
                  className="login__username-text"
                  id="username"
                  name="name"
                  onChange={[Function]}
                  placeholder="Phone number, username, or email"
                  type="text"
                />
              </label>
            </div>
            <div
              className="login__password"
            >
              <label
                className="login__password-label"
              >
                <input
                  className="login__password-text"
                  name="name"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </label>
            </div>
            <div
              className="login__submit"
            >
              <label>
                <button
                  aria-label={null}
                  className="login__submit-btn login__submit-btn-active btn btn-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Login In
                </button>
              </label>
            </div>
          </form>
          <div
            className="login__forget"
          >
            <div
              className="login__line"
            >
              <div
                className="login__line-abstract"
              />
              <div
                className="login__text-or"
              >
                or
              </div>
              <div
                className="login__line-abstract"
              />
            </div>
            <div
              className="login__forget-text"
            >
              Forgot password?
            </div>
          </div>
        </div>
        <div
          className="login__signup"
        >
          <p>
            Don't have an account? Sign up
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
